src/sentry/api/endpoints/relay/project_configs.py

```diff
@@ -62,16 +62,18 @@ def _post(self, request: Request):
             # Always compute the full config. It's invalid to send partial
             # configs to processing relays, and these validate the requests they
             # get with permissions and trim configs down accordingly.
-            response.update(self._post_or_schedule_by_key(request))
+            response.update(
+                # XXX(iker): we assume we can safely parse the version by
+                # casting it to an int if `_should_post_or_schedule` returns True
+                self._post_or_schedule_by_key(request, version=int(version))
+            )
         elif version in ["2", "3", "4"]:
             response["configs"] = self._post_by_key(
-                request=request,
-                full_config_requested=full_config_requested,
+                request=request, full_config_requested=full_config_requested, version=int(version)
             )
         elif version == "1":
             response["configs"] = self._post_by_project(
-                request=request,
-                full_config_requested=full_config_requested,
+                request=request, full_config_requested=full_config_requested, version=int(version)
```

Comment:
`_post_by_project` is legacy code for version 1. So instead of passing a version into this function, why not hard code the version inside the function body?